### ISSUE TITLE
[BUG][typescript-fetch] Emitted code returns a `string` for a `boolean` endpoint

### ISSUE THREAD

**Issue by cwboden** [OpenAPITools/openapi-generator#12788]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
For a simple server-side endpoint that returns a `boolean`, the generated code returns the value as a `string` 

##### openapi-generator version
```
openapi-generator-cli 6.0.1
  commit : 9e412ed
  built  : 2022-07-03T16:21:27Z
  source : https://github.com/openapitools/openapi-generator
  docs   : https://openapi-generator.tech/
```

##### OpenAPI declaration file content or url
```yaml
---
openapi: 3.0.1
info:
  title: OpenAPI definition
  version: v0
servers:
- url: http://foobar.com
  description: Generated server url
paths:
  "/my/server/endpoint":
    get:
      operationId: myServerEndpoint
      parameters:
      - name: capability
        in: query
        required: true
        schema:
          type: string
          enum:
          - CAPABILITY_FOO
          - CAPABILITY_BAR
      responses:
        '200':
          description: OK
          content:
            "*/*":
              schema:
                type: boolean
        '400':
          description: Bad Request
          content:
            "*/*":
              schema:
                type: object
                additionalProperties:
                  type: object
        '404':
          description: Not Found
          content:
            "*/*":
              schema:
                type: object
                additionalProperties:
                  type: object
        '500':
          description: Internal Server Error
          content:
            "*/*":
              schema:
                type: object
                additionalProperties:
                  type: object
```

##### Generation Details
Use the above OpenAPI declaration file and use the generator to produce the output with `typescript-fetch`. Then view the emitted code (or try using it with your endpoint)

##### Steps to reproduce
```bash
java -jar .\openapi-generator-cli.jar generate -i .\simplest-repro.yml -g typescript-fetch
```
(Or equivalent, based on what you saved the spec as)

##### Related issues/PRs
Tangentially related for encoding instead of decoding: #12449 

##### Suggest a fix
The issue is in the use of `TextApiResponse` for `myServerEndpointRaw()`. 

`TextApiResponse.value()` returns a `Promise<string>`, which we cast to `any` and assume it will automatically be coerced into `Promise<boolean>` in `myServerEndpointRaw()` *(see code example below)*.

We should either ensure it is cast correctly in `myServerEndpoint()` before being returned or create and use a new `Response` type, like `BooleanApiResponse` or `PrimitiveApiResponse` which can return non-string values.

**`DefaultApi.ts`**
```typescript
    async myServerEndpointRaw(
        requestParameters: MyServerEndpointRequest, 
        initOverrides?: RequestInit | runtime.InitOverrideFunction
    ): Promise<runtime.ApiResponse<boolean>> {
        /* snipped for brevity */

        const response = await this.request({
            /* snipped for brevity */
        }, initOverrides);

        return new runtime.TextApiResponse(response) as any;
    }

    async myServerEndpoint(
        requestParameters: MyServerEndpointRequest, 
        initOverrides?: RequestInit | runtime.InitOverrideFunction
    ): Promise<boolean> {
        const response = await this.myServerEndpointRaw(requestParameters, initOverrides);
        return await response.value();
    }
```

**`runtime.ts`**
```typescript
export class TextApiResponse {
    constructor(public raw: Response) {}

    async value(): Promise<string> {
        return await this.raw.text();
    };
}
```

**Comment by wtrocki** [comment]
@cwboden Is this issue an feature request to support OpenAPI 3.1 wildcard content types.

Generator does handle unknown type ("*/*") as string and let developers to `reparse it` on the user space. I think this would be very good default behaviour. 


if your endpoint returns `true`|`false` I cannot think of any other content type than application/text to handle it.

**Comment by cwboden** [comment]
@wtrocki Not necessarily. More that there's type disagreement -- TypeScript will complain if I call `myServerEndpoint()` and try and use the result like a `string`, since the declaration states it's a `Promise<boolean>`. But the generated code, since it uses `TextApiResponse`, is _actually_ a `string`, so we can't treat the response like a `boolean` either.

The emitted endpoint code should either state that it returns `Promise<string>`, so that I don't have to throw away type information or we should make the endpoint _actually_ return a `boolean` instead of just claiming to.

**Comment by borispetrovdev** [comment]
Just encountered this issue myself, except with `number` instead of `boolean`. Any progress on a generic fix?

**Comment by cwboden** [comment]
Looks like this was fixed in 7.0.0+
